model/src/test/kotlin/utils/LicenseResolverTest.kt

```diff
@@ -0,0 +1,244 @@
+        )
+    )
+)
+
+private fun OrtResult.addScanResult(id: Identifier, scanResult: ScanResult): OrtResult {
+    val scanResults = scanner!!.results.scanResults
+    var scanResultsForId = scanResults.find { it.id == id }?.let {
+        scanResults.remove(it)
+        it.results
+    } ?: emptyList()
+
+    scanResults += ScanResultContainer(id, scanResultsForId + scanResult)
+
+    return copy(
+        scanner = scanner!!.copy(
+            results = scanner!!.results.copy(
+                scanResults = scanResults
+            )
+        )
+    )
+}
+
+private fun OrtResult.getVcsProvenance(id: Identifier): Provenance? =
+    getProject(id)?.let { Provenance(vcsInfo = it.vcsProcessed) }
+        ?: getPackage(id)?.let { Provenance(vcsInfo = it.pkg.vcsProcessed) }
+
+private fun OrtResult.getSourceArtifactProvenance(id: Identifier): Provenance? =
+    getPackage(id)?.let { Provenance(sourceArtifact = it.pkg.sourceArtifact) }
+
+private fun createEmptyOrtResult() = OrtResult(
+    Repository.EMPTY,
+    AnalyzerRun(
+        environment = Environment(),
+        config = AnalyzerConfiguration(ignoreToolVersions = true, allowDynamicVersions = true),
+        result = AnalyzerResult.EMPTY
+    ),
+    ScannerRun(
+        environment = Environment(),
+        config = ScannerConfiguration(),
+        results = ScanRecord(
+            scanResults = sortedSetOf(),
+            storageStats = AccessStatistics()
+        )
+    )
+)
+
+class LicenseResolverTest : WordSpec() {
+    private val nextId = AtomicInteger(1)
+    private lateinit var ortResult: OrtResult
+    private lateinit var packageConfigurationProvider: PackageConfigurationProvider
+
+    private fun createResolver() = LicenseResolver(ortResult, packageConfigurationProvider)
+
+    private enum class ProvenanceType { VCS, SOURCE_ARTIFACT }
+
+    private fun setupScanResult(id: Identifier, type: ProvenanceType, licenseFindings: Collection<LicenseFinding>) {
+        val scanResult = ScanResult(
+            provenance = when (type) {
+                ProvenanceType.SOURCE_ARTIFACT -> ortResult.getVcsProvenance(id)
+                else -> ortResult.getSourceArtifactProvenance(id)
+            }!!,
+            scanner = ScannerDetails.EMPTY,
+            summary = ScanSummary(
+                startTime = Instant.MIN,
+                endTime = Instant.MAX,
+                fileCount = 0,
+                licenseFindings = licenseFindings.toSortedSet(),
+                copyrightFindings = sortedSetOf<CopyrightFinding>(),
+                packageVerificationCode = "",
+                issues = emptyList()
+            )
+        )
+
+        ortResult = ortResult.addScanResult(id, scanResult)
+    }
+
+    private fun setupPackage(): Package {
+        val id = nextId.getAndIncrement().toString()
+        val pkg = Package.EMPTY.copy(
+            id = Identifier(id),
+            sourceArtifact = RemoteArtifact("file:///$id", Hash.NONE),
```

Comment:
Looks a bit weird to use a *remote* artifact with `file://`, which is more expected for something *local*.

Reply:
Done.